client/apollo/react/src/Modal/__tests__/Modal.test.tsx

```diff
@@ -0,0 +1,153 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { useRef } from "react";
+import { Modal, type ModalProps } from "../ModalApollo";
+import type { HasScrollState } from "../../utilities/hook/useHasScroll";
+
+vi.mock("../../utilities/hook/useHasScroll.ts", () => ({
+  useHasScroll: vi.fn().mockReturnValue({
+    vertical: false,
+    horizontal: false,
+  } satisfies HasScrollState),
+}));
+
+const onCancelMock = vi.fn();
+const onSubmitMock = vi.fn();
+const onCloseMock = vi.fn();
+
+const defaultProps: ModalProps = {
+  title: "Modal Title",
+  children: "My content modal",
+  open: true,
+  onCancel: onCancelMock,
+  onSubmit: onSubmitMock,
+  onClose: onCloseMock,
+};
+
+const ModalDemo = (props: ModalProps) => {
+  const ref = useRef<HTMLDialogElement>(null);
+
+  return <Modal {...{ ...props, ref }}>My content modal</Modal>;
+};
+
+const getDialog = () => screen.getByRole("dialog", { hidden: true });
+
+const expectModalOpened = async () => {
+  const dialog = getDialog();
+  await waitFor(() => {
+    expect(defaultProps.onCancel).not.toHaveBeenCalled();
+    expect(dialog).toHaveStyle(`display: block`);
+    expect(dialog).toHaveAttribute("open", "true");
+  });
```

Comment:
pas besoin de tout mettre dans le waitFor, il faut juste mettre dedans le expect que tu dois attendre : 
```suggestion
  await waitFor(() => {
    expect(defaultProps.onCancel).not.toHaveBeenCalled();
  });
  expect(dialog).toHaveStyle(`display: block`);
  expect(dialog).toHaveAttribute("open", "true");
```